### PR TITLE
Build Token on SDK 1.3.8

### PR DESCRIPTION
**Pull request by t-nelson**
I need to link Ledger app tests against SDK 1.3.8 _and the SPL Token program.  So Token needs to upgrade, too!

### CODE REVIEW

File: token-swap/program/src/processor.rs [hunk 1]
```diff
@@ -553,27 +555,29 @@ mod tests {
         amount: u64,
     ) -> ((Pubkey, Account), (Pubkey, Account)) {
         let token_key = pubkey_rand();
-        let mut token_account = Account::new(0, size_of::<SplMint>(), &program_id);
+        let mut token_account = Account::new(0, <SplMint as Pack>::LEN, &program_id);
```
Comment (CriesofCarrots) [inline token-swap/program/src/processor.rs:558]: `SplMint::get_packed_len()`?

Comment (t-nelson) [reply token-swap/program/src/processor.rs:558]: Oh! Ha!  I meant to go looking for something less awkward after I got it working..  Thanks!

Comment (t-nelson) [reply token-swap/program/src/processor.rs]: Done!

**Comment by t-nelson** [comment]
Sigh... why can't I reproduce these errors locally?

File: token-swap/program/src/processor.rs [hunk 2]
```diff
@@ -597,7 +600,7 @@ mod tests {
 
     #[test]
     fn test_initialize() {
-        let swap_key = pubkey_rand();
+        let swap_key = Pubkey::from_str("7eLu8C6ZGATFZFXgzyem5oQaWMr2qTUBoHAbAoEQ5EZi").unwrap();
```
Comment (jackcmay) [inline token-swap/program/src/processor.rs:603]: This looks like a temporary bandaid fix, yes?

Comment (t-nelson) [reply token-swap/program/src/processor.rs:603]: I dunno, I've always perffered tests to run on static data

Comment (jackcmay) [reply token-swap/program/src/processor.rs:603]: What I mean is the usage model if this is an program address is to find one that's off the curve and pass the nonce.  That's not happening here so the test isn't mirroring the use case

Comment (t-nelson) [reply token-swap/program/src/processor.rs:603]: Sure.  I'll do it in a follow up tomorrow, though.  I've been trying to get a four line diff into the ledger app repo for ~20hrs now and am running out of steam

**Review by joncinque — APPROVED** [review]
Looks good to me -- the hard-coded address will get fixed up in #388

**Pull request merged**